### PR TITLE
Support verifying JWT with an array of potential keys

### PR DESCRIPTION
**Pull request by gangstead**
Implements multiple key support in a non-breaking way for issue #205 

This could be released as a minor update since it does not break any existing functionality.

### CODE REVIEW

**Comment by nelsonic** [comment]
👀

**Comment by gangstead** [comment]
😉

**Comment by gangstead** [comment]
Any thoughts one way or the other @nelsonic ?

File: lib/index.js [hunk 1]
```diff
@@ -79,24 +84,34 @@ internals.implementation = function (server, options) {
...
-                else {
-                  return reply.continue({ credentials: credentials || decoded, artifacts: token });
+          var keys = (internals.isArray(key)) ? key : [ key ];
+          var keysTried = 0;
+
+          keys.some(function (key) {
+            JWT.verify(token, key, verifyOptions, function (err, decoded) {
+              if (err) {
+                keysTried++;
+                if (keysTried >= keys.length) {
```
Comment (nelsonic) [inline lib/index.js]: @gangstead I don't see a situation where `keysTried` _would_ be greater than `keys.length` ... ? 🤔

**Comment by nelsonic** [comment]
Happy to merge your PR and publish a new version as it's 100% backwards compatible and _others_ may find this functionality _useful_... 👍 
(_note: I'm aware that doing the `keys.some` might add a micro-second of processing time to `each` request so if others feel strongly about performance we could re-write this functionality to be a switch where the `Array` of keys code branch would be separate from main execution..._)

**Pull request merged**